src/Umbraco.Core/Services/ContentTypeSchemaService.cs

```diff
@@ -0,0 +1,87 @@
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Models.PublishedContent;
+using Umbraco.Cms.Core.PublishedCache;
+using Umbraco.Cms.Core.Strings;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Core.Services;
+
+/// <inheritdoc />
+internal sealed class ContentTypeSchemaService : IContentTypeSchemaService
+{
+    private readonly IContentTypeService _contentTypeService;
+    private readonly IMediaTypeService _mediaTypeService;
+    private readonly IPublishedContentTypeCache _publishedContentTypeCache;
+    private readonly IShortStringHelper _shortStringHelper;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ContentTypeSchemaService"/> class.
+    /// </summary>
+    public ContentTypeSchemaService(
+        IContentTypeService contentTypeService,
+        IMediaTypeService mediaTypeService,
+        IPublishedContentTypeCache publishedContentTypeCache,
+        IShortStringHelper shortStringHelper)
+    {
+        _contentTypeService = contentTypeService;
+        _mediaTypeService = mediaTypeService;
+        _publishedContentTypeCache = publishedContentTypeCache;
+        _shortStringHelper = shortStringHelper;
+    }
+
+    /// <inheritdoc/>
+    public IReadOnlyCollection<ContentTypeSchemaInfo> GetDocumentTypes()
+        => GetContentTypeSchemaInfos(PublishedItemType.Content, _contentTypeService.GetAll());
+
+    /// <inheritdoc/>
+    public IReadOnlyCollection<ContentTypeSchemaInfo> GetMediaTypes()
+        => GetContentTypeSchemaInfos(PublishedItemType.Media, _mediaTypeService.GetAll());
+
+    private List<ContentTypeSchemaInfo> GetContentTypeSchemaInfos(
+        PublishedItemType itemType,
+        IEnumerable<IContentTypeComposition> contentTypes)
+    {
+        List<ContentTypeSchemaInfo> result = [];
+
+        foreach (IContentTypeComposition contentType in contentTypes)
+        {
+            IPublishedContentType publishedContentType;
+            try
+            {
+                publishedContentType = _publishedContentTypeCache.Get(itemType, contentType.Alias);
+            }
+            catch
+            {
+                // Skip content types that fail to load from cache
```

Comment:
Is this an indication that something is wrong?  If so, and if throwing isn't appropriate, then would it be worth logging a warning?